### PR TITLE
Migrate to CircleCI 2.0

### PR DESCRIPTION
**Pull request by shakuzen**
Makes the approximate same logic as CircleCI 1.0 written in 2.0 format.

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
kicked to make it build. pull requests were off for this repo https://circleci.com/gh/openzipkin/brave-cassandra

**Comment by shakuzen** [comment]
Merp. Thanks for configuring that. I'll look at this again and restore the workflow stuff and look into other things (no test reports, it seems) after work today.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            - build
+          filters:
+            branches:
+              only: master
+      - publish_stable:
+          requires:
+            - build
+          filters:
+            branches:
+              ignore: /.*/
```
Comment (shakuzen) [inline .circleci/config.yml:124]: Without this, the job will run on all branches. See [this section of the documentation](https://circleci.com/docs/2.0/workflows/#git-tag-job-execution):

> 1. For a branch push unaffected by any filters, CircleCI runs the job.
> 2. For a tag push unaffected by any filters, CircleCI skips the job.

**Comment by shakuzen** [comment]
There are no tests run or reported during the build, but this is an existing issue we can tackle in a subsequent pull request.

**Pull request merged**